merkle/smt/writer_test.go

```diff
@@ -114,12 +114,16 @@ func TestWriterWrite(t *testing.T) {
 		{desc: "unaligned", upd: []NodeUpdate{{ID: tree.NewNodeID2("ab", 10)}}, wantErr: "unexpected depth"},
 		{desc: "dup", upd: []NodeUpdate{upd[0], upd[0]}, wantErr: "duplicate ID"},
 		{desc: "2-shards", split: 128, upd: []NodeUpdate{upd[0], upd[1]}, wantErr: "writing across"},
-		{desc: "get-err", acc: noopAccessor{get: errors.New("nope")}, upd: []NodeUpdate{upd[0]}, wantErr: "nope"},
-		{desc: "set-err", acc: noopAccessor{set: errors.New("nope")}, upd: []NodeUpdate{upd[0]}, wantErr: "nope"},
+		{desc: "get-err", acc: &testAccessor{get: errors.New("nope")}, upd: []NodeUpdate{upd[0]}, wantErr: "nope"},
```

Comment:
"nope" is slightly confusing.  
Do you mean "nope" this error is not used -in which case nil would be better
Or do you mean "nope, this operation will fail".
In either case, a slightly more descriptive error would be great.

Reply:
I mean "nope, this operation will fail".

Reply:
#1909